packages/components/carousels/SmallCarousel/useSmallCarousel.ts

```diff
@@ -0,0 +1,80 @@
+import { useMemo, useRef, useState } from "react";
+import { StyleSheet } from "react-native";
+import {
+  ICarouselInstance,
+  TCarouselProps,
+} from "react-native-reanimated-carousel";
+
+export const useSmallCarousel = (
+  props: TCarouselProps & { height: number },
+) => {
+  const {
+    width = 0,
+    height = 0,
+    style,
+    data,
+    loop: isLoop = true,
+    ...carouselProps
+  } = props;
+
+  const carouselRef = useRef<ICarouselInstance | null>(null);
+  const viewWidth = StyleSheet.flatten(style)?.width;
+  const step =
+    width && typeof viewWidth === "number" ? Math.floor(viewWidth / width) : 0;
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [isScrolling, setScrolling] = useState(false);
+
+  const onScrollEnd = (index: number) => {
+    setCurrentIndex(index);
+    // Prevent spamming buttons
+    setScrolling(false);
+  };
+
+  const onPressPrev = () => {
+    // No matter the carousel or items width, if it's the last press on Prev, we set "Reached the carousel's start"
+    if (!isLoop && currentIndex - step <= 0) {
+      carouselRef.current?.scrollTo({ index: 0, animated: true });
+    } else {
+      carouselRef.current?.prev({ count: step });
+    }
+  };
+  const onPressNext = () => {
+    // No matter the carousel or items width, if it's the last press on Next, we set "Reached the carousel's end"
+    if (!isLoop && currentIndex + step >= data.length) {
+      carouselRef.current?.scrollTo({ index: data.length - 1, animated: true });
+    } else {
+      carouselRef.current?.next({ count: step });
+    }
+  };
+
+  const isPrevButtonEnabled = useMemo(
```

Comment:
IMHO, these calculations are very very lights, so we can skip useMemo in this case to have smaller code. Just personal idea ^^ you can ignore this message without pb

Reply:
Yes, I removed `useMemo` usage